hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/main/java/org/apache/hadoop/yarn/client/cli/QueueCLI.java

```diff
@@ -174,6 +225,63 @@ private int listChildQueues(String parentQueueName) throws IOException, YarnExce
   }
 
   private void printQueueInfo(PrintWriter writer, QueueInfo queueInfo) {
+    String schedulerType = queueInfo.getSchedulerType();
+
+    if (StringUtils.equals("FairScheduler", schedulerType)) {
+      printFairSchedulerQueue(writer, queueInfo);
+    } else {
+      printQueue(writer, queueInfo);
+    }
+  }
+
+  /**
+   * Print Queue information of FairScheduler.
+   *
+   * @param writer PrintWriter.
+   * @param queueInfo Queue Information.
+   */
+  private void printFairSchedulerQueue(PrintWriter writer, QueueInfo queueInfo) {
+    if (queueInfo.getSchedulerType() != null) {
+      writer.print("Scheduler Name : ");
+      writer.println(queueInfo.getSchedulerType());
+    }
+    writer.print("Queue Name : ");
+    writer.println(queueInfo.getQueueName());
+    writer.print("\tWeight : ");
+    DecimalFormat df = new DecimalFormat("0.00");
+    writer.println(df.format(queueInfo.getWeight()));
+    writer.print("\tState : ");
+    writer.println(queueInfo.getQueueState());
+    writer.print("\tMinResource : ");
+    writer.println("<memory : " + queueInfo.getMinResourceMemory() +
```

Comment:
We are doing this operation a lot, we should have some function to generate the String.

Reply:
Thanks for your suggestion! I will refactor this part of the code.